pkg/options/options.go

```diff
@@ -136,6 +136,7 @@ func (o *Options) AddFlags(cmd *cobra.Command) {
 
 	o.cmd.Flags().BoolVar(&o.CustomResourcesOnly, "custom-resource-state-only", false, "Only provide Custom Resource State metrics (experimental)")
 	o.cmd.Flags().BoolVar(&o.EnableGZIPEncoding, "enable-gzip-encoding", false, "Gzip responses when requested by clients via 'Accept-Encoding: gzip' header.")
+	o.cmd.Flags().BoolVar(&o.EnableUnscheduledPodsFetching, "enable-unscheduled-pods-fetching", false, "This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of unscheduled pods is fetched from the Kubernetes API Server. This is experimental.")
```

Comment:
Thinking about it again, would it really make sense for any user to want to opt-out of collection metrics from unscheduled pods?

As a user, I would expect to collect the same metrics whether I am using uuid sharding, node sharding or even an unsharded installation of ksm. So based on that I am wondering if we should really make this a flag and instead just enable that by default when node sharding is enabled.

Reply:
I think the issue is that as soon as you node shard, the current code will only select node-bound pods. Therefore you would run an additional deployment of KSM with this option enabled specifically. I don't think there is an easy way in the current codebase for us to filter pods on the current node and unbound pods at the same time. But I will leave this to the experts, this was just my current understanding.

Reply:
Ah indeed you are correct, I forgot we had to differentiate it from the other shards.

Reply:
So we should still set it to true by default and disable when sharding?